### PR TITLE
Fix/1 xos timeout

### LINKED ISSUE #1: Handle XOS timeout gracefully #1
-Handle XOS timeout so an AttributeError isn't thrown

**Details**

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
ERROR: couldn't get https://xos.acmi.net.au/api/works/ with exception: HTTPSConnectionPool(host='xos.acmi.net.au', port=443): Read timed out. (read timeout=30)... retrying
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/code/app/api.py", line 831, in <module>
    xos_private_api.get_works()
  File "/code/app/api.py", line 396, in get_works
    self.save_works_lists(resource)
  File "/code/app/api.py", line 488, in save_works_lists
    works_json = self.get(resource, params).json()
AttributeError: 'NoneType' object has no attribute 'json'

### PR DESCRIPTION
**Pull request by samaSiriwardhanage**
Resolves #1 

XOS is taking more than 30 seconds to render on the API. To solve this issue add params to the debug output, increase the timeout to 60 seconds, and raise the right exception if it continues to fail.

### Acceptance Criteria
- [ ]Update the XOS timeout 

### Relevant design files
* None

### Testing instructions
*None

### CODE REVIEW

File: app/api.py [hunk 1]
```diff
@@ -365,6 +365,12 @@ def get(self, resource, params=None):
             ) as exception:
                 print(f'ERROR: couldn\'t get {endpoint} with exception: {exception}... retrying')
                 retries += 1
+                if retries == 3:
+                    raise exception 
+<<<<<<< Updated upstream
+
+=======
+>>>>>>> Stashed changes
```
Comment (sighmon) [inline app/api.py]:
```suggestion
```

@samaSiriwardhanage Looks like you have some extra text introduced here - so could you please remove that and do another commit/push?

File: app/api.py [hunk 2]
```diff
@@ -355,7 +355,7 @@ def get(self, resource, params=None):
         retries = 0
         while retries < 3:
```
Comment (sighmon) [inline app/api.py]: @samaSiriwardhanage Seeing we are referencing the number of retries `3` twice here, could you please convert that to a variable at the top of the file?

**Review by sighmon — CHANGES_REQUESTED** [review]
Looking good so far Sama - I've got a couple of suggestions for you to look into. Best, Simon.

File: app/api.py (hunk 1 above)
Comment (samaSiriwardhanage) [reply app/api.py]: @sighmon I removed the unwanted text from the file.

File: app/api.py (hunk 2 above)
Comment (samaSiriwardhanage) [reply app/api.py]: @sighmon I've change the retries variable

Comment (sighmon) [reply app/api.py]: @samaSiriwardhanage Sorry I wasn't clear - a good programming trait is not to repeat things in your code that are likely to be forgotten and cause errors. So in this case we're repeating the number `3` in these two lines:

1. `while retries < 3:`
1. `if retries == 3:`

So what might be a good thing to do is to replace the number `3` with a new variable so we aren't repeating ourselves anymore.

When we create a variable that won't change, we make it all UPPERCASE in Python. So at the top of the file with the other UPPERCASE variables, let's create one called: `XOS_RETRIES` and set it equal to `3`.

Do you want to give that a try? Call me if you'd like to pair program on it.

**Review by sighmon — CHANGES_REQUESTED** [review]
Thanks for removing the extra lines of code.

I've got another suggestion for you. Call me if you'd like to walk through it. :-)

File: app/api.py (hunk 2 above)
Comment (samaSiriwardhanage) [reply app/api.py]: @sighmon I've push the commits.

**Review by sighmon — APPROVED** [review]
Nice work! That all looks good to me. :-)
